File: pokerManagement/Models/HandHistory.swift
import Foundation
import SwiftData

@Model
final class HandHistory {
    var timestamp: Date
    var holeCards: [String]
    var communityCards: [String]
    var action: String
    var potSize: Double
    var reasoning: String?

    init(timestamp: Date = .now, holeCards: [String], communityCards: [String],
         action: String, potSize: Double, reasoning: String? = nil) {
        self.timestamp = timestamp
        self.holeCards = holeCards
        self.communityCards = communityCards
        self.action = action
        self.potSize = potSize
        self.reasoning = reasoning
    }
}

File: android/app/src/main/java/com/pokermanagement/data/models/HandHistory.kt
package com.pokermanagement.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "hand_history")
data class HandHistory(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val timestamp: Long = System.currentTimeMillis(),
    val holeCards: String = "",           // JSON array string e.g. "[\"As\",\"Kd\"]"
    val communityCards: String = "",      // JSON array string
    val action: String = "",
    val potSize: Double = 0.0,
    val reasoning: String = ""
)
